### PR TITLE
[PR #1537 Patch 8] Refine the AutoMM tutorial

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Highlighted motivation.
2. Adjusted the presentation order of multimodal and single modal training.
3. Simplified each section.
4. Added one API section.

### CODE REVIEW

File: docs/tutorials/text_prediction/automm.md [hunk 1]
```diff
@@ -1,16 +1,19 @@
-# AutoMM for Image, Text, and Tabular
+# AutoMMPredictor for Image, Text, and Tabular
 :label:`sec_automm_predictor`
 
-AutoMM is a flexible framework that can train neural networks on image, text, numerical, and categorical data. For either single modal or multimodal data, AutoMM supports both classification and regression tasks. This tutorial demonstrates AutoMM's several application scenarios:
+Are you tired of switching codebases or hacking code for different data modalities (image, text, numerical, and categorical data) and tasks (classification, regression, and more)? `AutoMMPredictor` provides a one-for-all solution. This tutorial demonstrates several application scenarios.
```
Comment (sxjscience) [inline docs/tutorials/text_prediction/automm.md:4]: Consider to change `one-for-all solution` to `one-stop shop for multimodal deep learning models`

File: docs/tutorials/text_prediction/automm.md [hunk 2]
```diff
@@ -153,241 +95,211 @@ train_data = train_data.sample(500, random_state=0)
...
+Let's use `AutoMMPredictor` to train a late fusion model including [CLIP](https://huggingface.co/openai/clip-vit-base-patch32), [swin_small_patch4_window7_224](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/swin_transformer.py#L65), [google/electra-small-discriminator](https://huggingface.co/google/electra-small-discriminator), a numerical MLP, and a fusion MLP. 
 
 
 ```{.python .input}
 from autogluon.text.automm import AutoMMPredictor
 predictor = AutoMMPredictor(label=label_col)
 predictor.fit(
     train_data=train_data,
-    config=config,
+    config={
```
Comment (sxjscience) [inline docs/tutorials/text_prediction/automm.md:141]: Would it make sense to have a default value for `config` ? 

```
    config={
        "model": "fusion_mlp_image_text_tabular",
        "data": "default",
        "optimization": "adamw",
        "environment": "default",
    },
```

File: docs/tutorials/text_prediction/automm.md [hunk 3]
```diff
@@ -153,241 +95,211 @@ train_data = train_data.sample(500, random_state=0)
...
+You can flexibly customize any hyper-parameter in `config` via the `hyperparameters` argument of `.fit()`. To access one hyper-parameter in `config`, you need to traverse from top-level keys to bottom-level keys and join them together with `.` For example, if you want to change the per GPU batch size to 16, you can set `hyperparameters={"env.per_gpu_batch_size": 16}`.
 
-### TIMM + Huggingface Transformers + More
-AutoMM maximizes the choices of image and text backbones by supporting [timm](https://github.com/rwightman/pytorch-image-models) and [huggingface transformers](https://github.com/huggingface/transformers). AutoMM uses MLP for numerical data but converts categorical data to text by default. 
+## APIs
+Besides `.fit()` and `.evaluate()`, `AutoMMPredictor` also provides other useful APIs.
```
Comment (sxjscience) [inline docs/tutorials/text_prediction/automm.md:269]: It's not a bad idea to mention that it adopts a similar API as TextPredictor and TabularPredictor. You may use. 

```
Besides `.fit()` and `.evaluate()`, `AutoMMPredictor` also provides other useful APIs that are similar to that in TextPredictor and TabularPredictor. You may refer to more details in `:ref:sec_textprediction_beginner`.
```

File: docs/tutorials/text_prediction/automm.md (hunk 1 above)
Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm.md:4]: Sure.

File: docs/tutorials/text_prediction/automm.md (hunk 3 above)
Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm.md:269]: Sure.

File: docs/tutorials/text_prediction/automm.md (hunk 2 above)
Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm.md:141]: Added.

**Comment by szha** [comment]
Job PR-1585-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/1/index.html

**Comment by szha** [comment]
Job PR-1585-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/2/index.html

**Comment by szha** [comment]
Job PR-1585-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/3/index.html

**Comment by szha** [comment]
Job PR-1585-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/4/index.html

**Comment by szha** [comment]
Job PR-1585-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/5/index.html

File: docs/tutorials/text_prediction/automm.md [hunk 4]
```diff
@@ -153,241 +95,187 @@ train_data = train_data.sample(500, random_state=0)
...
+You can flexibly customize any hyper-parameter in `config` via the `hyperparameters` argument of `.fit()`. To access one hyper-parameter in `config`, you need to traverse from top-level keys to bottom-level keys and join them together with `.` For example, if you want to change the per GPU batch size to 16, you can set `hyperparameters={"env.per_gpu_batch_size": 16}`.
 
-### TIMM + Huggingface Transformers + More
-AutoMM maximizes the choices of image and text backbones by supporting [timm](https://github.com/rwightman/pytorch-image-models) and [huggingface transformers](https://github.com/huggingface/transformers). AutoMM uses MLP for numerical data but converts categorical data to text by default. 
+## APIs
+Besides `.fit()` and `.evaluate()`, `AutoMMPredictor` also provides other useful APIs, similar to those in `TextPredictor` and `TabularPredictor`. You may refer to more details in `:ref:sec_textprediction_beginner`.
```
Comment (sxjscience) [inline docs/tutorials/text_prediction/automm.md:245]:
```
:ref:sec_textpredition_beginner should be :ref:`sec_textprediction_beginner`
```

**Review by sxjscience — APPROVED** [review]
One last comment on the reference to the other tutorials. Overall looks great!

File: docs/tutorials/text_prediction/automm.md (hunk 4 above)
Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm.md]: Done.

**Comment by szha** [comment]
Job PR-1585-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/6/index.html

**Comment by szha** [comment]
Job PR-1585-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/7/index.html

**Comment by szha** [comment]
Job PR-1585-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1585/8/index.html
